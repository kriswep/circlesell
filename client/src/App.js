import React, { Component } from 'react';
import {
  ApolloClient,
  gql,
  graphql,
  ApolloProvider,
  // createNetworkInterface, // for live server
} from 'react-apollo';
// for mocked server
import { makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools';
import { mockNetworkInterfaceWithSchema } from 'apollo-test-utils';

// import { typeDefs } from '../../src/schema';
import typeDefs from './schema';

import PostList from './PostList';
import './App.css';

// Tut from here https://dev-blog.apollodata.com/full-stack-react-graphql-tutorial-582ac8d24e3b?_ga=2.249254087.1991559453.1497035474-344072256.1490729773

// for mocked server
const schema = makeExecutableSchema({ typeDefs });
addMockFunctionsToSchema({ schema });
const mockNetworkInterface = mockNetworkInterfaceWithSchema({ schema });

const client = new ApolloClient({
  networkInterface: mockNetworkInterface,
});

// for live server
// const networkInterface = createNetworkInterface({
//  uri: 'http://localhost:3000/graphql',
// });
// const client = new ApolloClient({
//  networkInterface,
// });

const postListQuery = gql`
  query PostsListQuery {
    posts(limit:100,offset:5){
      id
      title
      text
    }
  }
`;

export const PostListWithData = graphql(postListQuery)(PostList);

// eslint-disable-next-line react/prefer-stateless-function
class App extends Component {
  render() {
    return (
      <ApolloProvider client={client}>
        <div className="App">
          <div className="App-header">
            <h2>Welcome to Apollo Blog</h2>
          </div>
          {/* <PostList
            data={{
              loading: false,
              posts: [
                { id: 1, title: 'title1', text: 'Text' },
                { id: 2, title: 'title2', text: 'Text 2' },
              ],
            }}
          />*/}
          <PostListWithData />
        </div>
      </ApolloProvider>
    );
  }
}

export default App;
