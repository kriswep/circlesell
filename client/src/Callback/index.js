import React from 'react';
import loading from './loading.svg';

const Callback = () => {
  const style = {}; // ...

  return (
    <div style={style}>
      <img src={loading} alt="loading" />
    </div>
  );
};

export default Callback;
