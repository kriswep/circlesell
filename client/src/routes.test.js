/* globals test expect jest */
// import { shallow } from 'enzyme';
import { Route } from 'react-router';
import makeMainRoutes, { handleAuthentication } from './routes';
import * as Auth from './Auth';
import App from './App';
import Home from './Home';
import Callback from './Callback';

import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

test('routes should define mainRoute', () => {
  const routes = makeMainRoutes();
  expect(routes).toMatchSnapshot();
});

test('routes should handle auth', () => {
  // without auth
  expect(
    handleAuthentication.bind(null, {}, { location: { hash: '' } }),
  ).not.toThrow();

  // with auth
  const mMock = jest.fn();
  expect(
    handleAuthentication.bind(
      null,
      { handleAuthentication: mMock },
      { location: { hash: 'access_token' } },
    ),
  ).not.toThrow();
  expect(mMock).toHaveBeenCalled();
});

test('renders correct routes', () => {
  Auth.default = jest.fn();

  const routes = makeMainRoutes();

  const wrapper = shallow(routes);
  const routesRender = wrapper.find(Route).reduce((prev, route) => {
    const routeProps = route.props();
    // eslint-disable-next-line no-param-reassign
    prev[routeProps.path] = routeProps.render;
    return prev;
  }, {});
  // { 'nurse/authorization' : NurseAuthorization, ... }
  expect(typeof routesRender['/']).toBe('function');
  expect(typeof routesRender['/home']).toBe('function');
  expect(typeof routesRender['/callback']).toBe('function');

  expect(routesRender['/']().type).toEqual(App);
  expect(routesRender['/home']().type).toEqual(Home);
  expect(routesRender['/callback']({ location: { hash: '' } }).type).toEqual(
    Callback,
  );
});
