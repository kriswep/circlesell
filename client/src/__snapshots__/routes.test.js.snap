// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes should define mainRoute 1`] = `
<BrowserRouter
  component={[Function]}
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div>
    <Route
      path="/"
      render={[Function]}
    />
    <Route
      path="/home"
      render={[Function]}
    />
    <Route
      path="/callback"
      render={[Function]}
    />
  </div>
</BrowserRouter>
`;
